components/script/dom/videotracklist.rs

```diff
@@ -0,0 +1,118 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::dom::bindings::cell::DomRefCell;
+use crate::dom::bindings::codegen::Bindings::VideoTrackListBinding::{self, VideoTrackListMethods};
+use crate::dom::bindings::inheritance::Castable;
+use crate::dom::bindings::reflector::reflect_dom_object;
+use crate::dom::bindings::root::Dom;
+use crate::dom::bindings::root::DomRoot;
+use crate::dom::bindings::str::DOMString;
+use crate::dom::eventtarget::EventTarget;
+use crate::dom::videotrack::VideoTrack;
+use crate::dom::window::Window;
+use dom_struct::dom_struct;
+
+#[dom_struct]
+pub struct VideoTrackList {
+    eventtarget: EventTarget,
+    tracks: DomRefCell<Vec<Dom<VideoTrack>>>,
+}
+
+impl VideoTrackList {
+    pub fn new_inherited(tracks: &[&VideoTrack]) -> VideoTrackList {
+        VideoTrackList {
+            eventtarget: EventTarget::new_inherited(),
+            tracks: DomRefCell::new(tracks.iter().map(|track| Dom::from_ref(&**track)).collect()),
+        }
+    }
+
+    pub fn new(window: &Window, tracks: &[&VideoTrack]) -> DomRoot<VideoTrackList> {
+        reflect_dom_object(
+            Box::new(VideoTrackList::new_inherited(tracks)),
+            window,
+            VideoTrackListBinding::Wrap,
+        )
+    }
+
+    pub fn len(&self) -> usize {
+        self.tracks.borrow().len()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.tracks.borrow().is_empty()
+    }
+
+    pub fn item(&self, idx: usize) -> Option<DomRoot<VideoTrack>> {
+        self.tracks
+            .borrow()
+            .get(idx)
+            .map(|track| DomRoot::from_ref(&**track))
+    }
+
+    pub fn selected_index(&self) -> Option<usize> {
+        self.tracks
+            .borrow()
+            .iter()
+            .position(|track| track.selected())
+    }
+
+    pub fn set_selected(&self, idx: usize, value: bool) {
+        if let Some(track) = self.item(idx) {
+            track.set_selected(value);
+        }
+        self.upcast::<EventTarget>().fire_event(atom!("change"));
```

Comment:
Per https://html.spec.whatwg.org/multipage/media.html#toggle-audio-track, this needs to be a queued task like https://github.com/servo/servo/blob/f1dd31f70440fa9c7a40525bd1e03eede568f74d/components/script/dom/htmlmediaelement.rs#L328-L334. Also, we should only queue the task if an actual change occurred.